sensory_jamming.py

```diff
@@ -0,0 +1,342 @@
+            self.movdirection = float(movdirection)
+                # float. Direction (in radians) towards which the agent flies.
+                # Must be between -pi and pi (asserted later).
+            self.stepsize = float(flightspeed * self.d_t) 
+                # float. Distance in meters covered by the agent in 1 time step
+            self.IPI = int(float(IPI * m.pow(10,-3)) / env.d_t)
+                # int. Agent's inter-pulse interval, converted in time steps.
+            self.speedsound = 340.29 
+                # float. Speed of sound at sea level in m/s.
+            self.max_timestore = float(max_hear_dist / self.speedsound)
+                # float. Time maximal, in seconds, during which a sound can 
+                # travel, before its intensity passes below the hearing 
+                # threshold of the agent.
+            self.ring_width = float(1) 
+                # float. Spatial difference between start and end of call (m).
+            self.hearhistory = []
+                # empty array, dimensions 3*unknown. Will store the 
+                # identification number of the agent who's call has been heard 
+                # by the focal bat; the time at which the call was emitted; &
+                # the time at which it has been heard by the focal agent.
+        
+            assert self.movdirection <= m.pi and self.movdirection >= -(m.pi), "'movdirection' must be in radians & comprised between -pi & pi."
+                # returns an error message if movdirection is not within [-pi;pi].
+    
+        def Movement(self):
+            self.newx = float(self.x + self.stepsize * m.cos(self.movdirection))
+                # calculate the new x coordinate according to:
+                # - the distance travelled over 1 time step.
+                # - the direction of the movement
+            self.newy = float(self.y + self.stepsize * m.sin(self.movdirection))
+                # calculate the new y coordinate according to:
+                # - the distance travelled over 1 time step.
+                # - the direction of the movement
+        
+            self.x = self.Boundaries(self.newx, self.boxsize[0])
+                # verify that the new x coordinate is within boundaries
+            self.y = self.Boundaries(self.newy, self.boxsize[1])
+                # verify that the new y coordinate is within boundaries
+        
+            self.xhistory = np.append(self.xhistory, self.x)
+                # store the newly calculated x in xhistory
+            self.yhistory = np.append(self.yhistory, self.y)
+                # store the newly calculated y in yhistory
+        
+        def Calling(self):
+            
+            self.calltest = float(self.timestep-self.firstcall)/float(self.IPI)
+                # calculate the theoretical number of calls since the 1st call:
+                # time since 1st call = current time - time of 1st call/IPI
+            
```

Comment:
this does make sense mathematically - and is a neat implementation. From an 'biologically intuitive' perspective - could you change the way the self.calltest works - so that it keeps track of only the last call?  eg. 

self.lastcalltime= *code which looks at the last call by focal agent*
self.calltest= (timestep - lastcalltime) / IPI